src/js/lib/monster.ui.js

```diff
@@ -1038,27 +1050,28 @@ define(function(require) {
 			// If I click on the End Date, I shouldn't be able to select a day before the starting date, and I shouldn't be able to select anything after "today"
 			// If I click on the Start date, I should be able to select any day between the 1/1/2011 and "Today"
 			function customRange(input) {
-				var dateMin,
+				var dateMin = inputStartDate.datepicker('getDate'),
 					dateMax;
 
 				if (input.id === 'endDate') {
-					dateMin = inputStartDate.datepicker('getDate');
-
-					var dateMaxRange = new Date(dateMin);
+					var minMoment = moment(dateMin);
 
 					// If monthly mode, just increment the month for the maxDate otherwise, add the number of days.
 					if (range === 'monthly') {
-						dateMaxRange.setMonth(dateMaxRange.getMonth() + 1);
+						minMoment.add(1, 'months').subtract(1, 'days');
 					} else {
-						dateMaxRange.setDate(dateMaxRange.getDate() + range);
+						minMoment.add((range - 1), 'days');
 					}
-					dateMaxRange.setDate(dateMaxRange.getDate() - 1);
 
 					// Set the max date to be as far as possible from the min date (We take the dateMaxRange unless it's after "today", we don't want users to search in the future)
-					dateMax = dateMaxRange > today ? today : dateMaxRange;
+					dateMax = minMoment.toDate();
```

Comment:
To reduce the code and remove some extra nesting:
```suggestion
					dateMax = minMoment.isAfter(today) ? today.toDate() : minMoment.toDate();
```

Then remove the following if block:

```javascript
					if (minMoment.isAfter(today)) {
						dateMax = today.toDate();
					}
```